### PR TITLE
Set up the Javalin RouteOverviewPlugin

### LINKED ISSUE #169: Add RouteOverview Javalin plugin
See https://github.com/UMM-CSci-3601/3601-iteration-template/commit/70b3d45449a486b19ed464ca795bdab50ee428bf

### PR DESCRIPTION
**Pull request by NicMcPhee**
This adds an "/api" route that is populated by Javalin's RouteOverviewPlugin. This doesn't give us a *ton* of information, but it does list all the supported routes.

It was really easy to do, so I suspect we should go ahead and merge this in?

My sense is that going with [the OpenAPI plugin](https://javalin.io/plugins/openapi) would be more informative, but that's not happening this week. I'm also not sure if the added info would be worth the extra complication that would come with it.

Closes #169

### CODE REVIEW

**Review by helloworld12321 — COMMENTED** [review]
Looks good! 🙂

**Review by helloworld12321 — APPROVED** [review]

**Review by helloworld12321 — CHANGES_REQUESTED** [review]
Wait, actually, would it be all right to use some path other than `/api`?

In later labs we use `/api` as the prefix for all server-side routes, which could be a bit confusing—especially if we're going to apply `RouteOverviewPlugin` to the other labs.

Would you be all right with naming this endpoint something like `/route-overview`?

File: server/src/main/java/umm3601/Server.java [hunk 1]
```diff
@@ -22,6 +23,10 @@ public static void main(String[] args) {
       // This tells the server where to look for static files,
       // like HTML and JavaScript.
       config.addStaticFiles(CLIENT_DIRECTORY, Location.EXTERNAL);
+      // This adds a Javalin plugin that will list all of the
+      // routes/endpoints that we add below on a page reachable
+      // via the "/api" path.
+      config.registerPlugin(new RouteOverviewPlugin("/api"));
```
Comment (helloworld12321) [inline server/src/main/java/umm3601/Server.java:29]:
```suggestion
      // via the "/route-overview" path.
      config.registerPlugin(new RouteOverviewPlugin("/route-overview"));
```

Comment (NicMcPhee) [reply server/src/main/java/umm3601/Server.java:29]: I would be totally fine with a change like that, and it probably would make things a little cleaner. Thanks for the suggestion!

**Review by helloworld12321 — APPROVED** [review]
Looks great, thanks! ☺️

**Comment by floogulinc** [comment]
Actually it really should be `/api`. In future labs only things at the path `/api` are routed to the server at all and the idea was to actually return something at that path instead of an error which is now what will happen.

**Pull request merged**